pkg/kubeapiserver/default_storage_factory_builder.go

```diff
@@ -115,12 +114,14 @@ type completedStorageFactoryConfig struct {
 
 // New returns a new storage factory created from the completed storage factory configuration.
 func (c *completedStorageFactoryConfig) New() (*serverstorage.DefaultStorageFactory, error) {
-	resourceEncodingConfig := resourceconfig.MergeResourceEncodingConfigs(c.DefaultResourceEncoding, c.ResourceEncodingOverrides)
+	// resourceEncodingConfig := resourceconfig.MergeResourceEncodingConfigs(c.DefaultResourceEncoding, c.ResourceEncodingOverrides)
 	storageFactory := serverstorage.NewDefaultStorageFactory(
 		c.StorageConfig,
 		c.DefaultStorageMediaType,
 		c.Serializer,
-		resourceEncodingConfig,
+		// NewCompatibilityResourceConfig is intended to obviate the need for
+		// manual overrides list by taking skew into account.
+		serverstorage.NewCompatibilityResourceConfig(),
```

Comment:
I think this only affects kubeapiserver types (not aggregator/apiextensions), which override the `EncodeVersioner` for their storages manually